### DISCUSSION TITLE
Draw Matrix from RAM without anti-aliasing

### DISCUSSION THREAD

**Discussion by samcarey** [emilk/egui#1916] [Q&A] [answered]
I'm trying to replicate the way MATLAB can take a matrix of floats and plot it with a colormap, but in `egui`.

I have an `nalgebra` `DMatrix<f32>` that is changing rapidly and that I am plotting every frame. My current strategy is roughly the following:
```rust
// First, I map the floats into a `Vec<u8>` of RGBA values using the `colorgrad` library.
let image = epaint::ColorImage::from_rgba_unmultiplied(<size>, rgba_vec.as_slice());
let texture_handle = egui::Context::load_texture("image", image));
PlotUi::image(PlotImage::new(texture_handle, ...));
```

This basically works, but there are a couple of problems:
- The `load_texture` docs specifically say not to run it every frame, but I am. It seems like it's really intended for loading static images. Is this a problem, or is there a better/more efficient way?
- The displayed image has anti-aliasing, which is not what I want, because I want to see clearly the original data of the input matrix. Is there a way to turn this off?

**Comment by emilk** [comment, marked as answer]
> * The `load_texture` docs specifically say not to run it every frame, but I am. It seems like it's really intended for loading static images. Is this a problem, or is there a better/more efficient way

If the image data changes each frame, then there is no way to get around having to upload the image to the GPU each frame. The warning is only for users with static images that should re-upload the same image every frame.

> The displayed image has anti-aliasing, which is not what I want, because I want to see clearly the original data of the input matrix. Is there a way to turn this off?

Yes, in the next release of egui you can specify a texture filter in `Context::load_texture`.
